projects/fsastorefrontlib/src/cms-components/agent/appointment-scheduling-confirmation/appointment-scheduling-confirmation.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { RoutingService, WindowRef } from '@spartacus/core';
+import { genericIcons } from '../../../assets/icons/generic-icons';
+
+@Component({
+  selector: 'cx-fs-appointment-scheduling-confirmation',
+  templateUrl: 'appointment-scheduling-confirmation.component.html',
+})
+export class AppointmentSchedulingConfirmationComponent implements OnInit {
+  constructor(
+    protected domSanitizer: DomSanitizer,
+    protected routingService: RoutingService,
+    protected winRef: WindowRef
+  ) {}
+
+  appointmentDate: string;
+  imageLink = this.domSanitizer.bypassSecurityTrustUrl(
+    genericIcons.appointmentConfirmationImage
+  );
+
+  ngOnInit() {
+    const dateState = this.winRef.nativeWindow.history.state.date ?? null;
```

Comment:
I don't see the need for nullish coalesing here. If history date is undefined then null is in this case more-less the same, or am I missing something?

Reply:
Then in ngOnInit you would have just:
```
    if (!this.appointmentDate) {
      this.routingService.go('/');
    } 
```